web/src/server_events_dispatch.js

```diff
@@ -577,6 +578,9 @@ export function dispatch_normal_event(event) {
                         if (sub) {
                             stream_data.update_stream_email_address(sub, rec.email_address);
                             stream_events.mark_subscribed(sub, rec.subscribers, rec.color);
+                            if (overlays.is_user_profile_modal_open()) {
+                                user_profile.update_user_profile_streams_list();
```

Comment:
If it's not too awkward, I recommend de-duplicating these two-liners into a function in `user_profile` called `maybe_update_user_profile_streams_list`.

Reply:
Yeah, this isn't really a workable level of duplication. I'll post an idea for an interface that can be a prep PR.